spec/active_record/annotate/file_spec.rb

```diff
@@ -149,4 +153,19 @@ class User < ActiveRecord::Base
       expect(file.relative_path).to eq('namespace/path.rb')
     end
   end
+
+  describe "Annotation Errors" do
+    it "Removes Old Annotation" do
+      file.annotate_with(["error"], configurator)
+      expect(file).to be_changed
+
+      file.write
+      new_file = ActiveRecord::Annotate::File.new(file.path)
+
+      ### Doesnt add new annotation to non-annotated file
+      new_file.annotate_with(["error"], configurator)
+      expect(new_file).not_to be_changed
+    end
```

Comment:
I don't get the idea of this test. You are re-annotating the file with the same annotation and then checking that it didn't change :) Also the title assumes that old annotation should get removed which is not what happens here.